plugins/httpapi/aci.py

```diff
@@ -0,0 +1,191 @@
+        #         pickle.dump(0, fi)
+        self.backup_hosts = self.set_backup_hosts()
+        #self.connection.set_option('persistent_command_timeout', 3)
+        if not self.backup_hosts:
+            # if self.connection._connected is True and self.params.get('host') != self.connection.get_option("host"):
+            #     self.connection._connected = False
+            if self.params.get('host') is not None:
+                self.connection.set_option("host", self.params.get('host'))
+
+            if self.params.get('port') is not None:
+                self.connection.set_option("port", self.params.get('port'))
+
+            if self.params.get('username') is not None:
+                self.connection.set_option("remote_user", self.params.get('username'))
+
+            if self.params.get('password') is not None:
+                self.connection.set_option("password", self.params.get('password'))
+
+            if self.auth is not None:
+                self.connection._auth = {'Cookie': '{0}'.format(self.auth)}
+                self.check_auth_from_private_key = {'Cookie': '{0}'.format(self.auth)}
+
+            if self.params.get('use_proxy') is not None:
+                self.connection.set_option("use_proxy", self.params.get('use_proxy'))
+
+            if self.params.get('use_ssl') is not None:
+                self.connection.set_option("use_ssl", self.params.get('use_ssl'))
+
+            if self.params.get('validate_certs') is not None:
+                self.connection.set_option("validate_certs", self.params.get('validate_certs'))
+
+            #Case2: Switch using private key to credential authentication when private key is not specified
+            # if self.auth is None and self.check_auth_from_private_key is not None:
+            #     self.login(self.connection.get_option("remote_user"), self.connection.get_option("password"))
+        else:
+            # with open('/tmp/hosts.pkl', 'rb') as fi:
+            #     self.host_counter = pickle.load(fi)
+            try:
+                self.connection.set_option("host", self.backup_hosts[self.host_counter])
+                self.connection.queue_message('vvvv', 'Initializing operation on host {0}'.format(self.connection.get_option('host')))
+            except IndexError:
+                pass
+        # Perform some very basic path input validation.
+        path = str(path)
+        if path[0] != '/':
+            msg = 'Value of <path> does not appear to be formated properly'
+            raise ConnectionError(self._return_info(None, method, path, msg))
+        response = None
+        try:
+            response, rdata = self.connection.send(path, json, method=method)
+            self.connection.queue_message('vvvv', 'Received response from {0} with HTTP: {1}'.format(self.connection.get_option('host'), response.getcode()))
+            return self._verify_response(response, method, path, rdata)
+        except Exception:
```

Comment:
send_request() function is not complete yet because I need more APIC sims.
However, I will be adding more exceptions if we don't want the exceptions to be handled by handle_error function.